### PR TITLE
chat fixes v3 or 5 or 123

### LINKED ISSUE #140: Some mobile handling in the UI is wonky
- [ ]  When you start typing and keyboard comes up the layout breakpoint changes and breaks the layout on small sizes.
- [x]  Emoji picker is hiding at the bottom of the screen.
- [ ]  On iPad auto-jumping to bottom of chat does not reach the very bottom, it's ~1 message off.

### PR DESCRIPTION
**Pull request by gingervitis**
Made a bunch of various fixes to the Chat system. Hopefully I didn't break anything.

- kept a border below the video, on small screens (now that we've hidden the status info)
- some positioning things that may help #140 , such as making the input container not-fixed position, so that the end of the messages container doesn't actually hide behind the input container.
- fix max character counting bugs onPaste (addresses #154 ?)
- fix some typing problems when max chars reached, scubas not being able to command+a to select all and delete, or command+arrow to highlight text to delete
- hide emoji picker if max chars reached
- removed "Chat" button to save room
- made a custom emoji picker icon and put in the chat box to save space
- chat box grows with newlines (up to a max height)
- on mobile, when the keyboard pops up to type in chat, I'm hiding the header so you can see more messages (for smaller phones). This also fixes  #140 where layout may break when keyboard pops up.
- make Emoji picker show up on small mobile screens again
- totally unnecessary style updates to the Emoji picker
- address #101 by not jumping messages to bottom if the messages div has scrolled > 200px. Some of this might address (#158 )
- fix some horizontal text overflow by break-wording (#157)

### CODE REVIEW

**Comment by gabek** [comment]
I'm seeing a weird flicker in the text input when putting an emoji in it.  The emoji is taller than the default height of the text input, so it grows a little bit, but then starts changing height randomly.

Untitled

**Comment by gingervitis** [comment]
Hmm interesting, I can't repro on chrome nor safari. Does this happen consistently? 

> I'm seeing a weird flicker in the text input when putting an emoji in it. The emoji is taller than the default height of the text input, so it grows a little bit, but then starts changing height randomly.
> 
> Untitled

**Comment by gabek** [comment]
Not consistently.  It's random.  It'll flicker three times then stop for 5 seconds and then do it once and stop, for example.

It's more obvious in that animated GIF because of the low frame rate.  I tried to attach a video of it, but github won't allow video attachments.

**Comment by gingervitis** [comment]
what browser are you using?

**Comment by gabek** [comment]
Just my standard Brave browser:

```
> window.navigator.userAgent
"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.89 Safari/537.36"
```

**Comment by gabek** [comment]
Ok I can't reproduce it in Safari or my in a private tab.  So I'll chalk it up to being something specific in my browser.

**Comment by gabek** [comment]
Whoops didn't mean to close.

**Comment by gingervitis** [comment]
I made an update that might address the flickering you were seeing @gabek . I got it to happen when using keystrokes like Arrow or Command keys after inserting an emoji.

I also did end up removing the `requestAnimationFrame` things you put in last time for scrolling messages to the bottom. That actually added a noticeable delay in the message scrolling after typing. I decided to just try jumping to bottom with out "smooth" scrolling. Gets the job done faster.

File: webroot/js/utils/constants.js [hunk 1]
```diff
@@ -27,7 +27,26 @@ export const KEY_CHAT_FIRST_MESSAGE_SENT = 'owncast_first_message_sent';
...
+  'Delete',
+  'Backspace',
+];
+export const CHAT_KEY_MODIFIERS = [
+  'Control',
+  'Shift',
+  'Meta',
+  'Alt',
+];
+export const MESSAGE_JUMPTOBOTTOM_BUFFER = 260;
```
Comment (gingervitis) [inline webroot/js/utils/constants.js:49]: I'm finding that using such a fixed number doesn't work if the next message is really long. Might need to try the other option outlined in #101  🤔 

On a related note, should we cap the number of lines people can type in?

Comment (gabek) [reply webroot/js/utils/constants.js:49]: I have no opinion on if we should cap the number of lines.  I could see somebody being annoying by typing a single character per line for hundreds of lines, but it's probably rare.

**Comment by gabek** [comment]
> I made an update that might address the flickering you were seeing

Works great!!

**Comment by gabek** [comment]
I'm going to merge this in, throw it on goth.land, and play around with it.  But things have been good in local testing.  Exciting!!

**Pull request merged**